components/scream/src/physics/shoc/CMakeLists.txt

```diff
@@ -62,6 +62,14 @@ if (NOT CUDA_BUILD OR Kokkos_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE)
   ) # SHOC ETI SRCS
 endif()
 
+# Add dispatch source files if monolithic kernels are off
+if (NOT SCREAM_MONOLITHIC_KERNELS)
+  list(APPEND SHOC_SRCS
+    shoc_energy_integrals_disp.cpp
```

Comment:
Not sure if you want to do it in this PR, and it might be a personal taste thing, but in case you agree: the shoc (and p3) folders are quite big, given the large number of `_impl.cpp` and `.cpp` files (96 files in the shoc folder). This makes navigating the repo a bit hard (imho). I would vote for adding subfolders: one `impl` subfolder for the impl files, one `eti` subfolder for the ETI cpp files, and one `dispatch` for the new files you are adding. That will also allow the less experienced users to recognize the difference between eti and dispatch `*.cpp` files...

Reply:
I agree. This directory now has 98 files and will be growing to near 150 when this is done.